### PR TITLE
Sunodo deploy

### PR DESCRIPTION
**Pull request by tuler**
Provides a `sunodo deploy` command that performs the steps for a deployment of an application to a public networks. It's composed by the following tasks:

1. publishing of the cartesi machine
2. configuration of the network
3. configuration of the consensus protocol
4. configuration of the node reward mechanism

Initially it will be limited to:

1. IPFS only
2. some testnets, and local for dev mode
3. Authority only
4. Off-chain incentive only (ERC-20 left for a future release)

After the deployment the application is captured by a `sunodo controller` executed by some node provider.
The controller is part of #45

### CODE REVIEW

**Comment by tuler** [comment]
This is ready for a review.
Run `yarn build` and try the `sunodo deploy` command on a sunodo project.

File: apps/cli/src/deploy.ts [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+    if (status !== "success") {
+        throw new Error(`Error connecting to IPFS ${options.url}`);
+    }
+
+    const cid = await importDirectory(machine, options);
+    return { method: "ipfs", location: cid };
+};
+
+/**
+ *
```
Comment (endersonmaia) [inline apps/cli/src/deploy.ts:166]: Do we need a description here?

File: apps/cli/src/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+ * @param machine directory of cartesi machine snapshot to deploy
+ * @param options
+ * @returns selected publishing method and location of published machine
+ */
+const publish = async (
```
Comment (endersonmaia) [inline apps/cli/src/deploy.ts:171]: Not sure what's the usual (official?) term here, if publish or upload.

File: apps/cli/src/wallet.ts [hunk 3]
```diff
@@ -8,80 +8,146 @@ import {
...
+
+export const DEFAULT_DEVNET_MNEMONIC =
     "test test test test test test test test test test test junk";
 
+export type WalletType =
+    | "metamask"
+    | "walletconnect"
+    | "mnemonic"
+    | "private-key";
+const walletChoices = (chain: Chain): Choice<WalletType>[] => {
```
Comment (endersonmaia) [inline apps/cli/src/wallet.ts:62]: I could only test the mnemonic option.

Should we already have an AWS KMS option?

**Review by endersonmaia — APPROVED** [review]

File: apps/cli/src/deploy.ts (hunk 2 above)
Comment (tuler) [reply apps/cli/src/deploy.ts:171]: Publish meaning "make it public"

File: apps/cli/src/wallet.ts [hunk 4]
```diff
@@ -8,80 +8,146 @@ import {
...
+import { Choice, hexInput, selectAuto } from "./prompts.js";
+
+// list of all chains that will eventually be supported
+export const chains: Chain[] = [
+    arbitrum,
+    arbitrumGoerli,
+    mainnet,
+    optimism,
+    optimismGoerli,
+    sepolia,
```
[the remark this answers is not in the document]
Comment (tuler) [reply apps/cli/src/wallet.ts:38]: No, no need for goerli. It will sunset in the end of year.

**Comment by tuler** [comment]
@SocketSecurity ignore es5-ext@0.10.62

**Review by endersonmaia — APPROVED** [review]

**Pull request merged**